connectors/preflight_check.py

```diff
@@ -69,3 +71,26 @@ async def _check_system_indices_with_retries(self):
                     attempts += 1
                     await self._sleeps.sleep(self.preflight_idle)
         return False
+
+    def _check_configuration_validity(self):
+        # "Native" mode
+        configured_native_types = "native_service_types" in self.config
+        force_allowed_native = self.config.get("_force_allow_native", False)
```

Comment:
Does a key starting with `_` indicates `for internal use only`?

Will this be set to `True` for a cloud deployment?

I feel an `EnvVar` may be more appropriate if the answers to the above 2 questions are Yes.

Reply:
I'm not sure I see the strong benefit of env vars over yaml configuration. Most (all?) of our configuration is done through YAML - why should we divide it into multiple locations? You're right that we _could_, but I don't see the benefit.

Reply:
This certainly will work. But I just feel that the convention is to use `EnvVar` for some unpublished configuration. E.g. `NAME` and `PERF8` in `make ftest`:

https://github.com/elastic/connectors-python/blob/b45bb62b6e7be96196578189c1c08e81bdb09dfd/Makefile#L62

All fields in YAML should be configurable by users?

Reply:
Ah I see.
`ftest` being a separate bash tool, it makes sense that it needs to get configuration from outside of just our YAML file. But the connector service, when running in production, should be able to rely on YAML, I think.

> All fields in YAML should be configurable by users?

It, depends. In Enterprise Search we have undocumented YAML configs that we use for Feature Flags. In Elastic Cloud, some of our YAML configs are explicitly not accessible to the user (like encryption keys). I think we're ok to stick with this established pattern.